### DISCUSSION TITLE
Feature Request: Enable Symbol Palette

### DISCUSSION THREAD

**Discussion by simon4477** [yu-i-i/overleaf-cep#12] [Ideas]
It would be very nice to enable the Symbol Palette.

![image](https://github.com/user-attachments/assets/33ae9c47-3b00-4030-8571-2283a8d8c3e1)

**Comment by svdvonde** [comment]
This is an interesting one that I have briefly looked into. [Overleaf Ruhr](https://overleaf.lars.re/) claims to support it but I cannot confirm it because their deployments are behind a university login. One could try to deploy their instance from [the code](https://gitlab.fachschaften.org/newton/overleaf-delta) ... but I don't have time for it.

Instead I did a little dig in the code to see what they added to support the symbol palette. I found this in their commit history:

[Commit 3b4468b3](https://gitlab.fachschaften.org/newton/overleaf-delta/-/commit/3b4468b3c7b7ab0e7d775aef5f342001c12ccb39)
 * In [`server-ce/config/settings.js`](https://gitlab.fachschaften.org/newton/overleaf-delta/-/blob/main/server-ce/config/settings.js?ref_type=heads#L285) they added `symbolPalette: true`. 
 * In [`services/web/app/src/Features/Project/ProjectEditorHandler.js`](https://gitlab.fachschaften.org/newton/overleaf-delta/-/blob/main/services/web/app/src/Features/Project/ProjectEditorHandler.js?ref_type=heads#L62) they added `symbolPalette: true`. 

There were no other obvious changes related to the symbol palette that I could find, and which deviate from standard Overleaf, but admittedly I searched only using Gitlab's search feature and it's not great. I briefly tried to apply the 2 changes to a local (dev) instance of Overleaf, but it did not seem to give me a symbol palette. Further investigation is required..

**Reply by svdvonde** [reply]
I contacted the author of Overleaf Ruhr and they said:

> I unfortunately don't remember how I did this. Overleaf Ruhr will soon be superseded by a new, well-managed and maintained project called HajTeX. We want to have a first release in the coming weeks/months.

Someone will have to dig into the code of the current Overleaf to find out if it can be enabled or ported from an old version of Overleaf.

**Comment by yu-i-i** [comment]
The Symbol Palette was working with some minor tweaks in older versions before Overleaf switched from Angular to React back in May. I don’t think it’s a big deal to make it work again, but if I find time for Overleaf in the near future, I’d rather focus on template support: I remember symbols by heart :)

**Comment by yu-i-i** [comment]
Done.
